### PR TITLE
[PR] Adding content negotiation

### LINKED ISSUE #168: Use Content Negotiation to Return JSON from Endpoint
# Context

This project currently only allows one type of badge: [HitCount](http://hits.dwyl.com/dwyl/hits-issue-168)
This **meets _our_ needs** because this is the style we use in our repos e.g: [dwyl/fields](https://github.com/dwyl/fields)

But there are other people in the community who find _other_ badge designs more appealing.
We want to help them get what they need. 

The way for us to enable other badge designs is via Shields Endpoint: https://shields.io/endpoint

# Todo

+ [x] Follow: https://github.com/dwyl/phoenix-content-negotiation-tutorial
+ [x] Use: https://github.com/dwyl/content
+ [x] Enable the hits endpoint to return `.json` 
e.g: 
```json
{
  "schemaVersion": 1,
  "label": "hits",
  "message": "420",
  "color": "brightgreen"
}
```

That way people can apply whatever style/color they want to their badge.

### PR DESCRIPTION
**Pull request by LuchoTurtle**
closes #168 

This PR adds content negotiation so users can call [Shield.io's endpoint](https://shields.io/endpoint) and have customizable badges on their repositories.

### CODE REVIEW

**Comment by LuchoTurtle** [comment]
I'm just writing a small write-up in the README. Similarly to the rest of the README, I don't go into much detail.

**Comment by LuchoTurtle** [comment]
The tests should all pass and have 100% coverage. 
I think the CI has an outdated configuration, given the error.
```
Error: Tried to map a target OS from env. variable 'ImageOS' (got ubuntu22), but failed. If you're using a self-hosted runner, you should set 'env': 'ImageOS': ... to one of the following: ['ubuntu16', 'ubuntu18', 'ubuntu20', 'win16', 'win19']
```

**Comment by LuchoTurtle** [comment]
Finished adding the README. 
It should be good for review. The CI fails but it's setting up the VM. All tests should pass and the package should still have 100% coverage.
Perhaps changing CI on a different PR is prudent.

**Comment by nelsonic** [comment]
Trying to run this on my `localhost` getting:

```sh
mix c
ERROR! the application :mime has a different value set for key :types during runtime compared to compile time. Since this application environment entry was marked as compile time, this difference can lead to different behaviour than expected:

  * Compile time value was set to: %{}
  * Runtime value was not set

To fix this error, you might:

  * Make the runtime value match the compile time one

  * Recompile your project. If the misconfigured application is a dependency, you may need to run "mix deps.compile mime --force"

  * Alternatively, you can disable this check. If you are using releases, you can set :validate_compile_env to false in your release configuration. If you are using Mix to start your system, you can pass the --no-validate-compile-env flag

11:04:52.188 [error] Task #PID<0.220.0> started from #PID<0.95.0> terminating
** (stop) "aborting boot"
    (elixir 1.14.1) Config.Provider.boot/2
Function: &:erlang.apply/2
    Args: [#Function<1.104735216/1 in Mix.Tasks.Compile.All.load_apps/3>, [mime: "/Users/n/code/hits/_build/test/lib"]]
** (EXIT from #PID<0.95.0>) an exception was raised:
    ** (ErlangError) Erlang error: "aborting boot"
        (elixir 1.14.1) Config.Provider.boot/2
```

**Comment by nelsonic** [comment]
Ran:
```sh
MIX_ENV=test mix deps.compile mime --force && mix c
```
Got
```sh
==> mime
Compiling 1 file (.ex)
Generated mime app
Generated hits app

11:07:09.164 [info] Migrations already up
.............................
Finished in 0.4 seconds (0.1s async, 0.2s sync)
29 tests, 0 failures

Randomized with seed 698298
----------------
COV    FILE                                        LINES RELEVANT   MISSED
100.0% lib/hits.ex                                   154       17        0
100.0% lib/hits/application.ex                        40        4        0
100.0% lib/hits/hit.ex                                38        5        0
100.0% lib/hits/repo.ex                                5        0        0
100.0% lib/hits/repository.ex                         45        5        0
100.0% lib/hits/user.ex                               40        5        0
100.0% lib/hits/useragent.ex                          44        5        0
100.0% lib/hits_web/channels/hit_channel.ex           30        3        0
100.0% lib/hits_web/controllers/hit_controller.      209       42        0
100.0% lib/hits_web/controllers/page_controller       11        2        0
100.0% lib/hits_web/endpoint.ex                       50        0        0
100.0% lib/hits_web/gettext.ex                        24        0        0
100.0% lib/hits_web/router.ex                         33        5        0
100.0% lib/hits_web/views/error_helpers.ex            45        0        0
100.0% lib/hits_web/views/error_view.ex               16        1        0
100.0% lib/hits_web/views/hit_view.ex                  3        0        0
100.0% lib/hits_web/views/layout_view.ex               3        0        0
100.0% lib/hits_web/views/page_view.ex                 3        0        0
[TOTAL] 100.0%
----------------
Generating report...
Saved to: cover/
```

So far so good. 👌

File: README.md [hunk 1]
```diff
@@ -931,6 +931,152 @@ The comments are there for beginner-friendliness,
...
+which can later be used in 
+https://shields.io/endpoint.
+
+### Expected JSON response
+If you run `mix phx.server`
+and open a separate terminal session, 
+paste the following and run.
+
+```sh
+curl -H "Accept: application/json" http://localhost:4000/user/repo.svg\?color=blue
```
Comment (nelsonic) [inline README.md:1056]: Having the `.svg` in the URL and then `Accept: json` is contradictory. 
It's like saying "Double Cheeseburger Without Cheese" ... 🙄

Comment (LuchoTurtle) [reply README.md]: Should be fixed with [`2a23cfc` (#204)](https://github.com/dwyl/hits/pull/204/commits/2a23cfc415eb6f95d34219f23f61ecde31f331b8)

**Comment by nelsonic** [comment]
```
Unpacking objects: 100% (11/11), 1.27 KiB | 118.00 KiB/s, done.
remote: Total 11 (delta 7), reused 11 (delta 7), pack-reused 0
From github.com:dwyl/hits
   43d6cc8..2a23cfc  content-negotiation -> origin/content-negotiation
error: Your local changes to the following files would be overwritten by merge:
	README.md
Please commit your changes or stash them before you merge.
Aborting
```

**Comment by nelsonic** [comment]
This wastes my time. 😢

**Comment by nelsonic** [comment]
```
git pull
Auto-merging README.md
Auto-merging lib/hits_web/controllers/hit_controller.ex
CONFLICT (content): Merge conflict in lib/hits_web/controllers/hit_controller.ex
Automatic merge failed; fix conflicts and then commit the result.
```

**Comment by LuchoTurtle** [comment]
Want me to revert my commit?

**Comment by nelsonic** [comment]
@LuchoTurtle you've added a _lot_ of code to the `index/2` function in `lib/hits_web/controllers/hit_controller.ex`. ⏳ 
I've removed more than half of it and all tests still pass. ✂️ ✅ 
The merge conflicts are an epic time-suck. 😢

**Comment by nelsonic** [comment]
Generally when something is assigned to someone it's "theirs" to work on. 
That's what allows remote+async work. 
Otherwise in a larger team it's _instant_ chaos.

**Review by nelsonic — APPROVED** [review]
@LuchoTurtle nice work. Minor edits to enable `.json` but otherwise great! Thanks! 🎉

**Comment by nelsonic** [comment]
Custom badge

**Pull request merged**